tests/registry-e2e.rs

```diff
@@ -46,6 +47,40 @@ async fn expired_registrations_are_pruned_after_ttl() -> Result<(), anyhow::Erro
     Ok(())
 }
 
+#[tokio::test]
+async fn when_provider_registers_notifies_registry_observers() -> anyhow::Result<()> {
+    fn namespace_event(namespace: &str) -> String {
+        format!("namespaces/{}", namespace)
+    }
+
+    // arrange
+    let namespace = format!("e2e.registration.{}", get_uuid());
+
+    let builder = RegistrationBuilder::new(
+        "registration.provider.e2e",
+        "1.0.0",
+        "http://localhost:7090".parse().unwrap(), // arbitrary url, the provider will never be invoked
+        &namespace,
+        [Intent::Inspect],
+        ExecutionLocality::Local,
+    );
+
+    let mut subject = setup().await;
+
+    // act
+    let stream =
+        subject.listen("system.registry", vec![namespace_event(&namespace).into()]).await?;
+
+    builder.register_once(&mut None, true).await?;
+
+    // assert
+    let stream = stream.timeout(Duration::from_secs(5)).take(1).collect::<Vec<_>>().await;
+    let result = stream.into_iter().next().unwrap();
+    assert_eq!(namespace_event(&namespace).as_str(), result.unwrap().unwrap().id.as_ref());
```

Comment:
nit: you can remove all the `as_ref()`s and `into()` in the test code if you change the return type on the local helper:

```suggestion
    fn namespace_event(namespace: &str) -> Box<str> {
        format!("namespaces/{}", namespace).into()
    }

    // arrange
    let namespace = format!("e2e.registration.{}", get_uuid());

    let builder = RegistrationBuilder::new(
        "registration.provider.e2e",
        "1.0.0",
        "http://localhost:7090".parse().unwrap(), // arbitrary url, the provider will never be invoked
        &namespace,
        [Intent::Inspect],
        ExecutionLocality::Local,
    );

    let mut subject = setup().await;

    // act
    let stream = subject.listen("system.registry", vec![namespace_event(&namespace)]).await?;

    builder.register_once(&mut None, true).await?;

    // assert
    let stream = stream.timeout(Duration::from_secs(5)).take(1).collect::<Vec<_>>().await;
    let result = stream.into_iter().next().unwrap();
    assert_eq!(namespace_event(&namespace), result.unwrap().unwrap().id);
```